# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.h
# Commit: c942992fa981
# Full Hash: c942992fa981c66c424f9e16a50977529d4f6c54
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 10: Implement alertfinished behavior r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227695
# ==============================================================================

diff -r 11a3c2c86a8e -r c942992fa981 dom/notification/NotificationParent.h
--- a/dom/notification/NotificationParent.h	Mon Nov 11 20:34:14 2024 +0000
+++ b/dom/notification/NotificationParent.h	Mon Nov 11 20:34:15 2024 +0000
@@ -32,7 +32,9 @@
         mIsSecureContext(aIsSecureContext),
         mId(aId),
         mScope(aScope),
-        mOptions(aOptions) {};
+        mOptions(aOptions) {
+    MOZ_ASSERT(!mScope.IsEmpty(), "Only for persistent notifications for now");
+  };
 
   IPCResult RecvShow(ShowResolver&& aResolver);
 
@@ -47,6 +49,7 @@
 
   nsresult Show();
   nsresult FireClickEvent();
+  nsresult FireCloseEvent();
 
   void GetAlertName(nsAString& aRetval) {
     if (mAlertName.IsEmpty()) {
